CHEWBBACA/SchemaEvaluator/schema_evaluator.py

```diff
@@ -0,0 +1,703 @@
+import os
+import sys
+import json
+import statistics
+import pandas as pd
+import multiprocessing
+from operator import itemgetter
+from collections import Counter
+
+
+from Bio import SeqIO
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+from Bio.Align.Applications import MafftCommandline
+from Bio.Align.Applications import ClustalwCommandline
+
+try:
+    from utils import auxiliary_functions as aux
+except:
+    from CHEWBBACA.utils import auxiliary_functions as aux
+
+
+# Schema Evaluator Auxiliary Functions
+
+def gene_seqs_info_schema_evaluator(gene):
+    """ Determines the total number of alleles and the mean length
+        of allele sequences per gene.
+
+        Args:
+            genes_list (list): a list with names/paths for FASTA
+            files.
+        Returns:
+            genes_info (list): a list with a sublist for each input
+            gene file. Each sublist contains a gene identifier, the
+            total number of alleles for that gene and the mean length
+            of allele sequences for that gene.
+    """
+
+    seq_generator = SeqIO.parse(gene, "fasta")
+    alleles_lengths = [len(allele) for allele in seq_generator]
+    alleles_lengths.sort()
+
+    if len(alleles_lengths) == 0:
+        sys.exit("At least one file is empty or it doesn't exist. Exiting...")
+
+    # number of alleles
+    nr_alleles = len(alleles_lengths)
+
+    # minimum and maximum values
+    max_length = max(alleles_lengths)
+    min_length = min(alleles_lengths)
+
+    # Summary statistics
+    median_length = round(statistics.median(alleles_lengths))
+    mean_length = round(sum(alleles_lengths) / len(alleles_lengths))
+    mode_length = Counter(alleles_lengths).most_common()[0][0]
+
+    genes_info = [
+        gene,
+        nr_alleles,
+        min_length,
+        max_length,
+        median_length,
+        mean_length,
+        mode_length,
+        alleles_lengths
+    ]
+
+    return genes_info
+
+
+def gene_seqs_info_individual_schema_evaluator(gene):
+    """
+    """
+
+    seq_generator = SeqIO.parse(gene, "fasta")
+
+    # Allele IDs and lengths
+    allele_ids = []
+    alleles_lengths = []
+    for allele in seq_generator:
+        allele_ids.append(allele.id)
+        alleles_lengths.append(len(allele))
+
+    alleles_lengths.sort()
+
+    # number of alleles
+    nr_alleles = len(alleles_lengths)
+
+    # minimum and maximum values
+    max_length = max(alleles_lengths)
+    min_length = min(alleles_lengths)
+
+    # size range
+    size_range = f"{min_length}-{max_length}"
```

Comment:
f-Strings were introduced with Python 3.6. Will the code work with Python < 3.6?

Reply:
Well spotted @rfm-targa! If not such requirement must be specifically stated

Reply:
@rfm-targa you are correct, the code would not work with Python < 3.6. Will change to format in the next commit.